### PR TITLE
Omnia 1.1.0 release fixup

### PR DESCRIPTION
**Pull request by icetan**
- bump `seth` to `78508c6a8db2d6d3e8e09437dbe122bb5e6b2e7e`
- bump `setzer-mcd` to `103f8b0ee30d8e3f216be0365c20e12903d0629e`

### CODE REVIEW

**Comment by NiklasKunkel** [comment]
This branch was deployed on testnet and has been running stable. I did notice one issue though. Due to previous Coinmarketcap API being deprecated, `setzer` utilizes the new API which requires a registered API key. Setzer expects the  ENV `CMC_API_KEY` to contain this key. However, utilizing the nix installation/configuration process doesn't appear to have this key in scope. How do you suggest we best account for this?

**Comment by icetan** [comment]
> Setzer expects the ENV `CMC_API_KEY` to contain this key. However, utilizing the nix installation/configuration process doesn't appear to have this key in scope. How do you suggest we best account for this?

We could add the `CMC_API_KEY` environment variable to the `systemd` service config of `omnia`.

Or we could add it as a field in the omnia JSON config.

As the API key is required to run `omnia` correctly it might as well be placed in the domain of the `omnia` config.

What do you think @NiklasKunkel ?

Either way we would also need to add a option to the install script so that you can pass your private API key on install.

**Comment by NiklasKunkel** [comment]
I agree let's add it as a field to the `omnia` config. Given that users need to manually edit the `/etc/omnia.conf` the first time they install to put in `ethereum.from,`ethereum.keystore`, and `ethereum.password` I think it makes sense to have them add the CMC API key at that point.

>either way we would also need to add a option to the install script so that you can pass your private API key on install.

I think inputting things in multiple places is going to start getting really convoluted. Especially if not every user will run `omnia` through this mechanism. Let's keep it in the config for now. 

Can you confirm that exporting the ENV `CMC_API_KEY` inside `omnia` will make it in scope when `omnia` calls `setzer`? I believe it should, but don't understand the intricacies of `nix` enough to be sure.

**Comment by icetan** [comment]
> Given that users need to manually edit the /etc/omnia.conf the first time they install to put in ethereum.from, ethereum.keystore, and ethereum.password I think it makes sense to have them add the CMC API key at that point.

You can actually pass those values as options to the install script. So you do not have to edit `/etc/omnia.conf` to get running.

I think treating the API key in a similar fashion makes sense because you can install `omnia` with a oneliners.

> Can you confirm that exporting the ENV CMC_API_KEY inside omnia will make it in scope when omnia calls setzer? I believe it should, but don't understand the intricacies of nix enough to be sure.

Yes, it will work. Nix isn't a runtime, so same rules apply as when running things that are not in the Nix store.

It is common practise when packaging executables with Nix however to wrap them and unset certain environment variables, but this is done explicitly and we aren't doing this in the `setzer` Nix expression anywhere.

**Comment by NiklasKunkel** [comment]
>You can actually pass those values as options to the install script. So you do not have to edit /etc/omnia.conf to get running. I think treating the API key in a similar fashion makes sense because you can install omnia with a oneliners.

Neat, wasn't aware of that. 
Yes, in that case we should be consistent. Let's do both to allow for both types of users.

File: omnia/config.sh [hunk 1]
```diff
@@ -181,6 +183,20 @@ importOptionsEnv () {
 	[[ ${errors[*]} ]] && { printf '%s\n' "${errors[@]}"; exit 1; }
 }
 
+importServicesEnv () {
+	local _config="$1"
+	local _json
+
+	_json=$(jq -S '.services' < "$_config")
+	#extract cmc key
+	CMC_API_KEY="${CMC_API_KEY-$(jq -r '.cmcApiKey' <<<"$_json")}"
```
Comment (icetan) [inline omnia/config.sh:192]: You can remove the `_json` var and read the desired value directly from the config:

```suggestion
	CMC_API_KEY="${CMC_API_KEY-$(jq -r '.services.cmcApiKey' "$_config")}"
```

File: omnia/config.sh [hunk 2]
```diff
@@ -181,6 +183,20 @@ importOptionsEnv () {
...
+	#validate cmc key
+	[[ "$CMC_API_KEY" =~ ^[0-9a-f]{8}(-){1}[0-9a-f]{4}(-){1}[0-9a-f]{4}(-){1}[0-9a-f]{4}(-){1}[0-9a-f]{12}$ ]] || errors+=("Error - Coinmarketcap API Key is invalid.")
```
Comment (icetan) [inline omnia/config.sh:194]: `(-){1}` can be simplified to just `-`.

```suggestion
	[[ "$CMC_API_KEY" =~ ^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$ ]] || errors+=("Error - Coinmarketcap API Key is invalid.")
```

**Comment by NiklasKunkel** [comment]
Changes look good to me. We should eventually switch back to using `_json` in `importServices` as we add more propriatary data sources, but since CMC is the only service this works for now.

**Comment by icetan** [comment]
Just waiting for approval on this PR then.

**Comment by icetan** [comment]
Should we bump `ssb-server` to 15.2?

**Comment by NiklasKunkel** [comment]
Lets hold off on bumping ssb-server. It's very finicky. Changes around it need to be handled with extreme caution.

**Comment by NiklasKunkel** [comment]
Deployed on testnet, watched each Feed do a full cycle no issues, got a successful price update on the Ropsten ETHBTC Oracle.

**Pull request merged**